src/hypothesis/strategies.py

```diff
@@ -1155,23 +1155,36 @@ def decimals(min_value=None, max_value=None,
     # Set up a strategy for finite decimals
     if places is not None:
         # Fixed-point decimals are basically integers with a scale factor
+        # and annoying tendencies to break in unusual precision contexts.
```

Comment:
nit: "break" isn't a very useful word here. Could you elaborate on what goes wrong? (e.g. lose precision?)

Reply:
I've deleted this line and added a more useful comment.  In short:

- We don't want external state (ie precision context) to influence our results, because that can often break invariants (eg if bounds are not representable in current precision).  
- We therefore do all our arithmetic on decimals using methods on a context object with sufficient precision.  

Calculating that precision is where all the `math.log10` calls come in, and some subtleties about which values can be negative or non-integer and what to do in those cases, but I digress.